### DISCUSSION TITLE
Avoid escaping <script> content

### DISCUSSION THREAD

**Discussion by devster31** [marp-team/marp#250] [Q&A] [answered]
I am including a script in my ppt, below a minimal example and its output.
```md
---
marp: true
---

<script>
var a = 10
var b = 11
console.log(a > b)
</script>
```
returns:
```html
<script>
var a = 10
var b = 11
console.log(a &gt; b)
</script>
```

I'd like to avoid converting the `>` to `&gt;`.
Any tips?

**Comment by yhatt** [comment]
Consideration of `<script>` tag seems to be a lost part of [Marp Core's HTML sanitizer](https://github.com/marp-team/marp-core/blob/main/src/html/html.ts) because Marpit framework and general CommonMark parsers will output the correct JS code without escapes. 

---

If you're using Marp CLI, you can apply workaround by reverting HTML sanitaization to markdown-it's default through the custom engine.

```javascript
// engine.js
module.exports = ({ marp }) => marp.use((md) => {
  const { rules } = new md.renderer.constructor();
  md.renderer.rules.html_block = rules.html_block;
})
```

```
marp --engine ./engine.js --html unsafe-markdown.md
```

**Comment by yhatt** [comment]
I've worked for providing a better solution at marp-team/marp-core#282.

In the latest Marp tools, you can disable encoding HTML special characters by wrapping script in HTML comment `<!-- -->`.

```html
<script>
<!--
var a = 10
var b = 11
console.log(a > b)
-->
</script>
```

It's available in Marp CLI v1.5.2 or Marp for VS Code v1.4.7.

**Comment by yhatt** [comment, marked as answer]
**UPDATE:** Marp Core v3.9.1, v4.0.1 and after versions are no longer required to wrap the script with the HTML comment.
